### DISCUSSION TITLE
Timeline integrity

### DISCUSSION THREAD

**Discussion by beholdnec** [bluesky-social/atproto#849] [Clarifications and "Why?"] [answered]
Hello,

I'm wondering how atproto handles timeline integrity. That is, how atproto ensures that every post a certain user posted is visible if you're browsing from another federated server.

For some background, Mastodon has a longstanding issue where if you browse a user's timeline, there could be gaps. This occurs because not all the posts the other user made are federated onto your own server. The only reliable way to view a foreign user's timeline with all the posts intact is to browse directly to the other server's website.

Mastodon's lack of timeline integrity means that replies under posts may be missing, and reposts and like-counts may be less than the true number.

Does atproto have a solution to prevent timeline gaps?

**Comment by pfrazee** [comment, marked as answer]
Yeah in short, this isn't an issue. Information gets placed in repositories that are synced in a pull-based model. This makes delivery failures much less likely than a push-based model, since a pull sync can quickly establish missing records and get completely up-to-date. Views on data are produced by aggregators and you can further tune what's visible, both by changing your aggregator and by applying filters on the data they produce.

**Comment by beholdnec** [comment]
Thanks! Does atproto have a way to ensure private or follower-only posts aren't leaked to users who shouldn't be able to see them? Mastodon's solution is to send follower-only posts directly to the users, not to the public timeline.

**Reply by pfrazee** [reply]
We don't currently support private/follower-only posts. We're talking about how we can do it but I don't want to share something that ends up being wrong and confusing folks!

**Reply by ziiw** [reply]
New here,

Could it be done via a purpose made client?
